src/test/java/com/diffblue/deeptestutils/mock/MethodMatcherTest.java

```diff
@@ -0,0 +1,277 @@
+    // verify that we find the correct method according to Java language specification
+    Assert.assertEquals(child.argSubtype(child).toString(), s1);
+    // verify that the matched method returns what we expect it to return
+    Assert.assertEquals("5", s1);
+
+    String s3 = m3.invoke(child, parent).toString();
+    String s4 = m4.invoke(child, parent).toString();
+
+    // verify that we find the same method using both DTUMemberMatcher methods
+    Assert.assertEquals(s3, s4);
+    // verify that we find the correct method according to Java language specification
+    Assert.assertEquals(child.argSubtype(parent).toString(), s3);
+    // verify that the matched method returns what we expect it to return
+    Assert.assertEquals("4", s3);
+
+    String s5 = m5.invoke(parent, child).toString();
+    String s6 = m6.invoke(parent, child).toString();
+
+    // verify that we find the same method using both DTUMemberMatcher methods
+    Assert.assertEquals(s5, s6);
+    // verify that we find the correct method according to Java language specification
+    Assert.assertEquals(parent.argSubtype(child).toString(), s5);
+    // verify that the matched method returns what we expect it to return
+    Assert.assertEquals("4", s6);
+  }
+
+  @org.junit.Test
+  public void argSupertypeReturnSubtypeTest() throws IllegalAccessException, InvocationTargetException {
+    String name = "argSupertypeReturnSubtype";
+    Method m1 = DTUMemberMatcher.method(Child.class, name, Child.class);
+    Method m2 = DTUMemberMatcher.methodUsingWholeSignature(Child.class, name, Child.class);
+    Method m3 = DTUMemberMatcher.method(Child.class, name, Parent.class);
+    Method m4 = DTUMemberMatcher.methodUsingWholeSignature(Child.class, name, Parent.class);
+
+    Child child = new Child("child");
+    Parent parent = new Parent("parent");
+
+    String s1 = m1.invoke(child, child).toString();
+    String s2 = m2.invoke(child, child).toString();
+
+    // verify that we find the same method using both DTUMemberMatcher methods
+    Assert.assertEquals(s1, s2);
+    // verify that we find the correct method according to Java language specification
+    Assert.assertEquals(child.argSupertypeReturnSubtype(child).toString(), s1);
+    // verify that the matched method returns what we expect it to return
+    Assert.assertEquals("6", s1);
+
+    String s3 = m3.invoke(child, parent).toString();
+    String s4 = m4.invoke(child, parent).toString();
+
+    // verify that we find the same method using both DTUMemberMatcher methods
+    Assert.assertEquals(s3, s4);
+    // verify that we find the correct method according to Java language specification
+    Assert.assertEquals(child.argSupertypeReturnSubtype(parent).toString(), s3);
+    // verify that the matched method returns what we expect it to return
+    Assert.assertEquals("7", s3);
+  }
+
+  @org.junit.Test
+  public void mix1Test() throws IllegalAccessException, InvocationTargetException {
+    String name = "mix1";
```

Comment:
:pick: Some tests define the string in a variable, some not. Maybe make that consistent (consider defining these strings as constant class variables).